store/tikv/2pc.go

```diff
@@ -411,6 +302,28 @@ func (c *twoPhaseCommitter) primary() []byte {
 	return c.primaryKey
 }
 
+func (c *twoPhaseCommitter) secondaries() [][]byte {
+	if c.primaryIdx < 0 {
+		return [][]byte{}
+	}
+	// Easy case, primary key is first.
+	if c.primaryIdx == 0 {
+		return c.mutations.keys[1:]
+	}
+
+	// Hard case, primary key is in the middle somewhere, need to copy the array and skip the primary key.
+	secondaries := make([][]byte, len(c.mutations.keys)-1)
```

Comment:
For the stripped keys with `Op_CheckNotExists` key op type, they are sent by `prewrite` requests but ignored by `commit`. Maybe it should not affect the crash recovery and `commitTS` calculation, do we need to add these keys into secondaries?

Reply:
I think we don't need to check them